### ISSUE TITLE
Code requires "allowSyntheticDefaultImports" TypeScript option

### ISSUE THREAD

**Issue by wereHamster** [apollographql/apollo-server#2025]
Otherwise you get errors like this:

```
node_modules/apollo-server-express/dist/ApolloServer.d.ts:1:8 - error TS1192: Module '"…/node_modules/@types/express/index"' has no default export.

1 import express from 'express';
         ~~~~~~~

node_modules/apollo-server-express/dist/ApolloServer.d.ts:2:8 - error TS1192: Module '"…/node_modules/@types/cors/index"' has no default export.

2 import corsMiddleware from 'cors';
         ~~~~~~~~~~~~~~
```

You're supposed to import these modules with `import * as express from 'express'` instead. If you don't, you need to enable the allowSyntheticDefaultImports TypeScript option.

**Comment by eugene1g** [comment]
Related #1977 

It might be not a bad technical requirement (seems to be 'recommended' by TS), but definitely a breaking change.

**Comment by abernix** [comment]
Relates-to or duplicated-by #1977, #2519, #2521 (and more).

Thanks so much for opening this originally.  The important thing here is that the consuming project needs to adopt `esModuleInterop: true`.  While enabling `esModuleInterop` within Apollo Server was more of a breaking change than intended (or even understood, at the time it was done), it's been almost a year now since it was first enabled.  This flag first became available in TypeScript 2.7 and helps align TypeScript with the behavior chosen by Babel, Webpack, Rollup, React Native, etc.  It's been the default configuration generated by `tsc --init` for quite some time and at this point we're intending to stick with `esModuleInterop: true` going forward.

It _may_ be possible to avoid the type errors on an existing project by enabling `allowSyntheticDefaultImports` (which doesn't change the code emitted, just changes the type-checking behavior), but I haven't verified.
